features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Create/GraphQL/Steps/ProvideGraphQL.jsx

```diff
@@ -0,0 +1,277 @@
+        return (
+            <React.Fragment>
+                <React.Fragment>
+                    {valid.graphQLFile.invalidFile && (
+                        <div className={classes.errorMessageWrapper}>
+                            <ErrorOutline className={classes.errorIcon} />
+                            <Typography variant='body2' gutterBottom className={classes.errorMessage}>
+                                {errorMessage}
+                            </Typography>
+                        </div>
+                    )}
+                    <Dropzone
+                        onDrop={this.handleUploadFile}
+                        multiple={false}
+                        className={classNames(classes.dropZone, {
+                            [classes.dropZoneErrorBox]: valid.graphQLFile.empty,
+                        })}
+                    >
+
+                        {file.name ? (
+                            <div>
+                                <Icon className={classes.largeIcon}>
+                                    insert_drive_file
+                                </Icon>
+                                <div className={classes.fileNameWrapper}>
+                                    <Typography variant='body2' gutterBottom>
+                                        {file.name} - {file.size} bytes
+                                    </Typography>
+                                </div>
+                            </div>
+                        ) : (
+                            <div>
+                                <Backup className={classes.dropZoneIcon} />
+                                <div>
+                                    <FormattedMessage
+                                        id='try.dropping.schema.file.here.or.click.to.select.schema.to.upload'
+                                        defaultMessage='Try dropping schema file here, or click to select
+                                        schema to upload.'
+                                    />
+                                </div>
+                            </div>)}
+
+                    </Dropzone>
+                    <FormHelperText className={classes.errorMessage}>
+                        {valid.graphQLFile.empty && (
+                            <FormattedMessage id='error.empty' defaultMessage='This field can not be empty.' />
+                        )}
+                        {valid.graphQLFile.invalidFile && (
+                            <FormattedMessage id='error.invalid.graphql.file' defaultMessage='Invalid GraphQL File' />
+                        )}
+                    </FormHelperText>
+                </React.Fragment>
+            </React.Fragment>
+        );
+    }
+}
+
+ProvideGraphQL.defaultProps = {
+    file: {},
+};
+
+ProvideGraphQL.propTypes = {
+    updateGraphQLBean: PropTypes.func.isRequired,
+    validate: PropTypes.bool.isRequired,
+    valid: PropTypes.shape({}).isRequired,
+    file: PropTypes.shape({}),
```

Comment:
we will need to add the parameters that are called with file as well for example 

`
valid: PropTypes.shape({
name:PropTypes.string.isRequired
}).isRequired
`

Reply:
Fixed